packages/nestjs/src/backtrace.interceptor.ts

```diff
@@ -0,0 +1,155 @@
+
+    /**
+     * If specified, matching errors will not be sent.
+     * Can be an array of error types, or a function returning `boolean`.
+     * The error is passed to the function as first parameter.
+     *
+     * @example
+     * // Exclude BadRequestException or errors deriving from it
+     * {
+     *   excludeExceptionTypes: [BadRequestException]
+     * }
+     *
+     * // Exclude errors that are instanceof BadRequestException
+     * {
+     *   excludeExceptionTypes: (error) => error instanceof BadRequestException
+     * }
+     */
+    readonly excludeExceptionTypes?: ExceptionTypeFilter;
+}
+
+/**
+ * Intercepts errors and sends them to Backtrace.
+ */
+@Injectable()
+export class BacktraceInterceptor implements NestInterceptor {
+    private readonly _client?: BacktraceClient;
+
+    /**
+     * Creates an interceptor with the global client instance.
+     *
+     * The instance will be resolved while intercepting the request.
+     * If the instance is not available, an error will be thrown.
+     */
+    constructor(options?: BacktraceInterceptorOptions);
+    /**
+     * Creates an interceptor with the provided client instance.
+     */
+    constructor(options: BacktraceInterceptorOptions | undefined, client: BacktraceClient);
+    constructor(
+        private readonly _options = BacktraceInterceptor.getDefaultOptions(),
+        @Optional() client?: BacktraceClient,
+    ) {
+        this._client = client;
+    }
+
+    public intercept(context: ExecutionContext, next: CallHandler<unknown>): Observable<unknown> {
+        const client = this._client ?? BacktraceClient.instance;
+        if (!client) {
+            throw new Error('Backtrace instance is unavailable. Initialize the client first.');
+        }
+
+        return next.handle().pipe(
+            catchError((err) => {
+                if (!this.shouldSend(err)) {
+                    return throwError(() => err);
+                }
+
+                const attributes = this.getAttributes(context);
+
+                if (typeof err !== 'string' && !(err instanceof Error)) {
+                    client.send(String(err), attributes);
+                } else {
+                    client.send(err, attributes);
+                }
+
+                return throwError(() => err);
+            }),
+        );
+    }
+
+    private shouldSend(error: unknown) {
+        if (this._options.includeExceptionTypes && !this.filterException(error, this._options.includeExceptionTypes)) {
+            return false;
+        }
+
+        if (this._options.excludeExceptionTypes && this.filterException(error, this._options.excludeExceptionTypes)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private getAttributes(context: ExecutionContext) {
+        const type = context.getType();
+        switch (type) {
+            case 'http':
+                return this.getHttpAttributes(context.switchToHttp());
```

Comment:
can we extract anything interesting from those attributes? Like for example HTTP request URL, HTTP method, controller name, etc?

Reply:
done